libs/common/src/main/java/org/opensearch/common/annotation/InternalApi.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Internal APIs that have no compatibility guarantees and should be not used outside
+ * of OpenSearch core components.
+ *
+ * @opensearch.api
+ */
+@Documented
+@Target({ ElementType.TYPE, ElementType.PACKAGE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+@PublicApi(since = "2.10.0")
+public @interface InternalApi {
```

Comment:
Apologies if this has been discussed elsewhere and I missed it, but according to the documentation above:

> any class explicitly marked with the `@InternalApi` (or documented as `@opensearch.internal`) annotation, or not explicitly marked by an annotation

which raises the question of why have this annotation at all, given that it is equivalent to "not explicitly marked by an annotation"?

Reply:
For me, the "not explicitly marked" language was there to cover this interim phase where 90% of the code isn't explicitly marked. Ideally going forward we want everything explicitly marked because "not explicitly marked" raises questions of intent.

Reply:
Thanks @andrross , have your concerns been cleared out?

Adding to that (from developer point of view): when looking into source code, seeing something like `@InternalApi` catches the eye (vs having nothing).

Reply:
I consider this resolved, resolving!